app/models/concerns/image_file.rb

```diff
@@ -4,16 +4,57 @@ module ImageFile
   extend ActiveSupport::Concern
 
   included do
+    has_one :task, as: :owner
+
     before_validation :detect_image_attributes
 
-    validates :url, presence: true
+    before_validation :guid
```

Comment:
a little redundant since also called in #initialize_attributes, no?  -- unless you're thinking it's possible we'll have an image without a guid on a non-create ?